proposed-solutions.md

```diff
@@ -1,35 +1,65 @@
 ### 5. what are the proposed solutions?
 
-#### 1. double parsing
+#### 1. Double Parsing
 
-Under this approach, the file would be parsed twice.
-If an `import` or `export` statement is found, evaluate as an ES Module,
-or Node Module if there are none.
+Parse the file twice, accepting the first error-free result as the type of the
+module. One could either parse the file as a Script first or as a Module first.
 
-This is not currently possible because the spec is ambiguous,
-however it is possible that the spec may change in this regard.
+There are some issues with this approach:
 
-#### 2. directives ("use module")
+* Tooling authors have to implement a parse step in order to determine file
+  type. Not all languages have a JavaScript parser readily available.
+* It is valid to have an ES module with no `import` or `export` statements,
+  leading to the potential for ambiguity: if a file was intended to be
+  a Node.js module but parsed as an ES module, it could result different
+  `"use strict"` semantics being silently applied to the file, causing hard
+  to reason about behavior.
 
-This approach is actually possible, given that import statements must be top-level. (That is, not within evaluated code blocks.)
+#### 2. Directives (e.g., "use module")
 
-#### 3. magic bytes ("js:;")
+Determine the module type of the file by the presence or absence of a `"use
+module"` directive, defined and implemented *outside* of the JS specification.
 
-You write `js:;` at the beginning of a file to run as an ES Module.
+There are some issues with this approach:
 
-This eliminates the _major_ problem with the directives approach.
+* A circular problem arises: in order to accurately determine whether to parse
+  the file as an ES module or a Node.js module, one must first parse the file
+  looking for directive statements *(Consider that a directive may be proceeded
```

Comment:
also, might be worth making explicit that this requires AST extraction and that regexps are insufficient